lib/modules/dosomething/dosomething_signup/dosomething_signup.query.inc

```diff
@@ -38,23 +38,24 @@ function dosomething_signup_get_signups_query($params = []) {
  *   - campaigns: (string) Campaign nid(s) to filter by.
  * @return SelectQuery object
  */
-function dosomething_signup_build_signups_query($params) {
-  $query = new EntityFieldQuery();
-  $query
-    ->entityCondition('entity_type', 'signup');
+function dosomething_signup_build_signups_query($params = []) {
+  $query = db_select('dosomething_signup', 's');
+  $query->join('dosomething_reportback', 'rb', 's.nid = rb.nid AND s.run_nid = rb.run_nid');
+  $query->fields('s', ['sid', 'uid', 'nid', 'run_nid']);
+  $query->fields('rb', ['rbid', 'uid', 'nid', 'run_nid']);
```

Comment:
I think you can eliminate most of the fields in this call, since all we only need the `rbid` from the `rb` table, otherwise it's just duplicate data:

```
$query->fields('rb', ['rbid']);
```

Currently this is obtained for a result:

```
stdClass Object
(
    [sid] => 32
    [uid] => 12
    [nid] => 6
    [run_nid] => 1740
    [timestamp] => 1394231397
    [rbid] => 10
    [rb_uid] => 12
    [rb_nid] => 6
    [rb_run_nid] => 1740
)
```
and there's duplicate data, just with a different key name :)